### PR TITLE
feat: implement options to pass custom LLDB command and python script files…

### PR DESCRIPTION
**Pull request by gusc**
This is a re-submitted pull request from a separate feature branch.

I wrote a proposal in shazron/phonegap-questions#57

Here is the implementation of the first solution - two arguments that would add ability to pass custom LLDB commands and a custom Python script to handle communication with the process. This will require for the user to implement the whole "fruitstrap" logic on their own side, but it also gives way for custom communication with LLDB Python APIs which wasn't possible before.

### CODE REVIEW

**Comment by shazron** [comment]
I'll review soon.

**Comment by gusc** [comment]
OK, cool, just to note - there will be merge conflicts between this and #339, so after merging this I'll probably need to update the other.

**Comment by shazron** [comment]
Pushing to 1.9.4 since I don't have time to review for 1.9.3 (need to get fix out for build issue)

**Comment by shazron** [comment]
I merged #339 which had some lldb script changes. You mentioned that there might be conflicts, can you take a look.

**Comment by gusc** [comment]
Does anyone has any objections against this pull request? It's been hanging here for more than a year, I could merge it, but it would be for the best that someone at least checks if it does not break anythin.

File: ios-deploy.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -237,7 +255,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "echo \"\\\"# AUTO-GENERATED - DO NOT MODIFY\\\"\" > src/ios-deploy/lldb.py.h\nawk '{ print \"\\\"\"$0\"\\\\n\\\"\"}' src/scripts/lldb.py >> src/ios-deploy/lldb.py.h\n";
+			shellScript = "echo \"\\\"# AUTO-GENERATED - DO NOT MODIFY\\\\\\n\\\"\" > src/ios-deploy/lldb.py.h\nawk '{ print \"\\\"\"$0\"\\\\n\\\"\"}' src/scripts/lldb.py >> src/ios-deploy/lldb.py.h\n";
```
Comment (gabebear) [inline ios-deploy.xcodeproj/project.pbxproj]: Why add a newline here? It shouldn't cause any issues, but I'm wondering if this does anything.

**Comment by gabebear** [comment]
I'm weary of adding more odd shell scripts to our build system rather than using than using odd strings. This will likely break people who are trying to build with systems that aren't standard Xcode(e.g. bazel of ninja).

File: ios-deploy.xcodeproj/project.pbxproj (hunk 1 above)
Comment (gusc) [reply ios-deploy.xcodeproj/project.pbxproj]: Well it's been more than a year, I really don't remember :D

Comment (gabebear) [reply ios-deploy.xcodeproj/project.pbxproj]: Overall this PR seems fine... I'm mainly worried about having to update other build systems to include that extra shell script when it seems like we could not add it.

**Comment by gusc** [comment]
I don't see no new shell scripts - this just moves some LLDB commands out to external source (i.e. cleans up the main source file a bit) and adds a feature to add your own LLDB comands  to run when app is installed on the device.

But I agree that this should be tested with those exotic build systems.

File: ios-deploy.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -226,6 +231,19 @@
...
+			buildActionMask = 2147483647;
+			files = (
+			);
+			inputPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "echo \"\\\"# AUTO-GENERATED - DO NOT MODIFY\\\\\\n\\\"\" > src/ios-deploy/lldb_cmds.h\nawk '{ print \"\\\"\"$0\"\\\\n\\\"\"}' src/scripts/lldb_cmds >> src/ios-deploy/lldb_cmds.h\n";
```
Comment (gabebear) [inline ios-deploy.xcodeproj/project.pbxproj]: I know that other people are building this not using xcodebuild and that this change will require updating those build systems to create a system to emulate what this awk script does (adding escape sequences to src/scripts/lldb_cmds and placing it in src/ios-deploy/lldb_cmds.h).

**Comment by gabebear** [comment]
This PR is a bit stale... but is also a place where we could make ios-deploy better... The awk script python injection makes building ios-deploy janky... What about removing that? We can just include the source of the Python script and add the escaping when we inject it(i.e. remove the Python ShellScript in the Xcode project file).

Removing the Python ShellScript would remove my objections here.

**Comment by gabebear** [comment]
Closing this for now since XCode12 has Python script changes.

**Pull request closed without merging**